code/game/objects/structures/vehicles/vehicle.dm

```diff
@@ -405,6 +405,7 @@
 	if (loc == oldloc)
 		return
 	if(next_cart)
+		sleep(0)
```

Comment:
Why is this necessary now?

Reply:
Without `sleep(0)` the proc `next_cart.Move(oldloc)` fails when src has density (for example a closed locker is loaded). This means the cart stops pulling the next cart.

I can't guarantee why this happens, but I have a hypothesis: `src.Move(newloc)` doesn't stop processing when we reach this statement (probably due to some `sleep()` or `spawn()` on Move code), so `next_cart.Move(oldloc)` thinks `src` is still on `oldloc` and fails to move. `sleep(0)` comes into play and just makes sure we're on the back of the line of processing stuff for this tick.